### PR TITLE
RIA-7769 Fix HearingChannel and HearingLocation prepopulate.

### PR DESCRIPTION
**Pull request by MiguelMartinRojas**
### Jira link (if applicable)

https://tools.hmcts.net/jira/browse/RIA-7769

### Change description ###

RIA-7769 Fix HearingChannel and HearingLocation prepopulate.

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/RecordAdjournmentDetailsMidEventHandler.java [hunk 1]
```diff
@@ -34,15 +39,32 @@ public PreSubmitCallbackResponse<AsylumCase> handle(PreSubmitCallbackStage callb
...
-        asylumCase.read(LIST_CASE_HEARING_CENTRE, HearingCentre.class)
-            .ifPresent(hearingCentre -> asylumCase.write(NEXT_HEARING_LOCATION, hearingCentre));
+                .ifPresent(hearingLength -> asylumCase.write(NEXT_HEARING_DURATION, hearingLength));
+        asylumCase.read(HEARING_CENTRE, HearingCentre.class)
+                .ifPresent(hearingCentre -> asylumCase.write(NEXT_HEARING_LOCATION, hearingCentre));
+        return responseCheckingHearingRange(asylumCase);
+    }
 
-        return new PreSubmitCallbackResponse<>(asylumCase);
+    private PreSubmitCallbackResponse<AsylumCase> responseCheckingHearingRange(AsylumCase asylumCase) {
```
Comment (robin-ieong) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/RecordAdjournmentDetailsMidEventHandler.java:57]: minor one but could use a more meaningful name for this method like validateHearingDateRange

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/RecordAdjournmentDetailsMidEventHandler.java [hunk 2]
```diff
@@ -34,15 +39,32 @@ public PreSubmitCallbackResponse<AsylumCase> handle(PreSubmitCallbackStage callb
...
+        PreSubmitCallbackResponse<AsylumCase> preSubmitCallbackResponse = new PreSubmitCallbackResponse<>(asylumCase);
+        asylumCase.read(NEXT_HEARING_DATE, String.class).ifPresent(nextHearingDate -> {
+            if (nextHearingDate.equals("ChooseADateRange")) {
```
Comment (robin-ieong) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/RecordAdjournmentDetailsMidEventHandler.java:60]: This looks like something that should be within an enum

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/RecordAdjournmentDetailsMidEventHandler.java [hunk 3]
```diff
@@ -34,15 +39,32 @@ public PreSubmitCallbackResponse<AsylumCase> handle(PreSubmitCallbackStage callb
...
+                if (asylumCase.read(NEXT_HEARING_DATE_RANGE_EARLIEST, String.class).isEmpty()
+                        && asylumCase.read(NEXT_HEARING_DATE_RANGE_LATEST, String.class).isEmpty()) {
+                    preSubmitCallbackResponse.addError("You must provide one of the earliest or latest hearing " +
```
Comment (robin-ieong) [inline src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/RecordAdjournmentDetailsMidEventHandler.java:63]: error messages should ideally be constants

File: src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/RecordAdjournmentDetailsMidEventHandler.java (hunk 2 above)
Comment (MiguelMartinRojas) [reply src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/RecordAdjournmentDetailsMidEventHandler.java:60]: I will create a constant for it.

**Review by robin-ieong — APPROVED** [review]

**Review by rajeshthuraiyur — APPROVED** [review]
LGTM

**Pull request merged**